examples/src/lib.rs

```diff
@@ -166,3 +195,37 @@ pub enum ExampleType {
         num_signers: usize,
     },
 }
+
+/// Defines a set of available hash function for STARK protocol.
+///
+/// Choice of a hash function has a direct impact on proof generation time, proof size, and proof
+/// soundness. In general, sounds of the proof is bounded by the collision resistance of the hash
+/// function used by the protocol.
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum HashFunction {
+    /// BLAKE3 hash function with 192 bit output.
+    ///
+    /// When this function is used in the STARK protocol, proof security cannot exceed 96 bits.
+    Blake3_192,
+
+    /// BLAKE3 hash function with 256 bit output.
+    ///
+    /// When this function is used in the STARK protocol, proof security cannot exceed 128 bits.
+    Blake3_256,
+
+    /// SHA3 hash function with 256 bit output.
+    ///
+    /// When this function is used in the STARK protocol, proof security cannot exceed 128 bits.
+    Sha3_256,
+
+    /// Rescue Prime hash function with 248 bit output. It only works in `f62` field.
+    ///
+    /// When this function is used in the STARK protocol, proof security cannot exceed 124 bits.
+    Rp62_248,
```

Comment:
Let's get rid of this option. I'm not sure it adds much given we have `Rp64_256` supported.

Reply:
done